### PR TITLE
SST Guns

### PR DESCRIPTION
**Pull request by MSieghardt**
**Soteria Surface Tension Project**
With amazing sprites from HOUND

Introducing the SST (Soteria Surface Tension) guns, which are the result of a collaboration between Soteria Medical and Soteria Research. Four guns that use Greyson tech to synthesize their projectiles, coat them in soporific and then fire them. They run off batteries, but unlike the Guild railguns, they don't pack as much of a stopping punch, instead having greater power efficiency.
The Abnegate: A .35-like handgun that is good enough to take and subdue targets while not going overboard.
The Format Bound: A 40-like handgun that packs a hardier punch and delivers more soporific at the cost of fire speed and energy cost.
The Humility: Semi-automatic shotgun with soporific coated synthesized beanbags. A true painbringer.
The System Cost: A light machinegun that fires soporific coated rubber bullets, albeit with less delivered chemicals than the other weapons in the SST project. 

Non-Lethal, Orderly-like end-game weapons for when you have to say "That's it, corpsman, I warned you" in Soteria.

### CODE REVIEW

File: code/modules/projectiles/projectile/bullettypes.dm [hunk 1]
```diff
@@ -89,6 +99,15 @@
...
 	step_delay = 0.9
 
+/obj/item/projectile/bullet/magnun_40/rubber/soporific
+	name = "soporific coated rubber bullet"
+
+/obj/item/projectile/bullet/magnun_40/rubber/soporific/on_hit(atom/target, def_zone = null)
+	if(isliving(target))
+		var/mob/living/L = target
+		if(istype(L) && L.reagents)
+			L.reagents.add_reagent("stoxin", 5)
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/projectile/bullettypes.dm:109]:
```suggestion
			L.reagents.add_reagent("stoxin", 3)
```

File: code/modules/projectiles/projectile/bullettypes.dm [hunk 2]
```diff
@@ -22,6 +22,15 @@
...
 	sharp = FALSE
 
+/obj/item/projectile/bullet/pistol_35/rubber/soporific
+	name = "soporific coated rubber bullet"
+
+/obj/item/projectile/bullet/pistol_35/rubber/soporific/on_hit(atom/target, def_zone = null)
+	if(isliving(target))
+		var/mob/living/L = target
+		if(istype(L) && L.reagents)
+			L.reagents.add_reagent("stoxin", 3)
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/projectile/bullettypes.dm:32]:
```suggestion
			L.reagents.add_reagent("stoxin", 2)
```

File: code/modules/projectiles/projectile/bullettypes.dm [hunk 3]
```diff
@@ -250,6 +269,15 @@
...
 	step_delay = 0.9
 
+/obj/item/projectile/bullet/rifle_75/rubber/soporific
+	name = "soporific coated rubber bullet"
+
+/obj/item/projectile/bullet/rifle_75/rubber/soporific/on_hit(atom/target, def_zone = null)
+	if(isliving(target))
+		var/mob/living/L = target
+		if(istype(L) && L.reagents)
+			L.reagents.add_reagent("stoxin", 2)
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/projectile/bullettypes.dm:279]:
```suggestion
			L.reagents.add_reagent("stoxin", 1)
```

File: code/modules/research/nodes/greyson.dm [hunk 4]
```diff
@@ -60,8 +60,21 @@
...
 	unlocks_designs = list(/datum/design/research/item/greyson/glass_widow)
+/datum/technology/SST
+	name = "Soteria Surface Tenstion Project"
+	desc = "The development of non-lethal weapons for Soteria Medical using GP technology."
+	x = 0.3
+	y = 0.3
+	tech_type = RESEARCH_GREYSON
+	icon = "abnegate"
+	required_technologies = list(/datum/technology/GP_window)
+	required_tech_levels = list(RESEARCH_COMBAT = 10)
```
Comment (Trilbyspaceclone) [inline code/modules/research/nodes/greyson.dm:72]:
```suggestion
```
Cleans up code a bit

File: code/modules/research/nodes/greyson.dm [hunk 5]
```diff
@@ -60,8 +60,21 @@
 								 /datum/technology/exotic_gunmods)
 	required_tech_levels = list(RESEARCH_COMBAT = 10)
 	cost = 15000
-
 	unlocks_designs = list(/datum/design/research/item/greyson/glass_widow)
+/datum/technology/SST
```
Comment (Trilbyspaceclone) [inline code/modules/research/nodes/greyson.dm:64]:
```suggestion

/datum/technology/sst
```

File: code/modules/projectiles/guns/energy/sst.dm [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+	charge_cost = 150
+	gun_tags = list(GUN_PROJECTILE, GUN_LASER, GUN_ENERGY)
+	suitable_cell = /obj/item/weapon/cell/medium
+	fire_delay = 12
+	recoil_buildup = 6
+	can_dual = FALSE
+	damage_multiplier = 1.5
+	projectile_type=/obj/item/projectile/bullet/magnun_40/rubber/soporific
+	matter = list(MATERIAL_PLASTEEL = 18, MATERIAL_STEEL = 10,  MATERIAL_SILVER = 12, MATERIAL_PLATINUM = 5)
+	price_tag = 2000
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/guns/energy/sst.dm:40]:
```suggestion
	price_tag = 1600
```

File: code/modules/projectiles/guns/energy/sst.dm [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+	icon_state = "format_bound"
+	item_state = "format_bound"
+	fire_sound = 'sound/weapons/guns/fire/pistol_fire.ogg'
+	charge_cost = 150
+	gun_tags = list(GUN_PROJECTILE, GUN_LASER, GUN_ENERGY)
+	suitable_cell = /obj/item/weapon/cell/medium
+	fire_delay = 12
+	recoil_buildup = 6
+	can_dual = FALSE
+	damage_multiplier = 1.5
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/guns/energy/sst.dm:37]:
```suggestion
	damage_multiplier = 1.3
```

**Comment by MSieghardt** [comment]
@Trilbyspaceclone suggestions commited. We should squeeze this before merging of course.

**Pull request merged**